Samples/AudioPreprocessing/WinUI/AudioPreprocessing/AudioPreprocessing/Model/PreprocessingModel.cs

```diff
@@ -0,0 +1,259 @@
+﻿using NAudio.Wave;
+using System;
+using System.Collections.Generic;
+using Microsoft.AI.MachineLearning.Experimental;
+using Microsoft.AI.MachineLearning;
+using Windows.Media;
+using Windows.Graphics.Imaging;
+
+using Operator = Microsoft.AI.MachineLearning.Experimental.LearningModelOperator;
+using System.Diagnostics;
+
+namespace AudioPreprocessing.Model
+{
+    public class PreprocessModel
+    {
+        const string MS_EXPERIMENTAL_DOMAIN = "com.microsoft.experimental";
+
+        public string AudioPath { get; set; }
+
+        public string MelSpecImagePath { get; set; }
+
+        public PreprocessModel()
+        {
+            AudioPath = "..\\tmp\\mel_spectrogram_file.jpg";
+            MelSpecImagePath = "..\\tmp\\mel_spectrogram_file.jpg";
+        }
+
+        public SoftwareBitmap GenerateMelSpectrogram(string audioPath)
+        {
+            var signal = GetSignalFromFile(audioPath);
+            int batchSize = 1;
+            int signalSize = signal.Count;
+            int windowSize = 256;
+            int dftSize = 256;
+            int hopSize = 3;
+            int nMelBins = 1024;
+            int samplingRate = 8192;
+            var rawSoftwareBitmap = GetMelspectrogramFromSignal(signal, batchSize, signalSize, windowSize, dftSize,
+                hopSize, nMelBins, samplingRate);
+
+            return rawSoftwareBitmap;
+        }
+
+        private List<float> Resample(List<float> signal, int oldSampleRate, int newSampleRate)
```

Comment:
Is this the best way to resample? A low pass filter may be needed here for better results right? (could possibly accomplish this via more ops in the graph...

Reply:
Or maybe this is just temporary to get things working? If so can we add a comment?

Reply:
Yes! This was a temporary function that I used at one point to debug the inputs to the graph. Removed